src/05-many-two-state-systems.md

```diff
@@ -0,0 +1,768 @@
+
+```python
+j_index[1]
+```
+
+Again much like tutorial 4, the next step is to extract the states in rows/columns `[0,1,2]` from the operators that make up the Hamiltonian (i.e. the $J$'s).
+
+
+We'll make a function to automate this process. It takes two arguments:
+1. The number of TSS we want to simulate
+2. The value of $j$ we want to simulate
+
+```python
+def make_operators(num_tss, j):
+    
+    j_index, jm_list = j_states_list(num_tss)
+    
+    # Check that the j we want to simulate is a valid j for this number of TSS.
+    # If we don't choose a valid j then return a helpful list of valid options
+    try:
+        j_index[j]
+    except:
+        raise Exception(f"j needs to be one of {j_vals(num_tss)}")
+    
+    # Create the J operator for the number of TSS we have specified
+    J = jspin(num_tss)
+    
+    # Now we extract only the states (and labels for those states) for the j we have specified 
+    J = [Ji.extract_states(j_index[j]) for Ji in J]  # Loops through the 3 J components and does `extract` for each one
+    jm_list = [jm_list[i] for i in j_index[j]]  
+    
+    
+    return J, jm_list
+```
+
+We are ready. Let's re-simulate fig 4.
+
+```python
+J, jm_list = make_operators(2, 1)
+```
+
+```python
+result, ekets = simulate(A, delta, J, times, 0)
+```
+
+```python
+psi, P = make_p_psi_arrays(result.states, ekets) 
+```
+
+```python
+df = pd.DataFrame(P, index=times)
+```
+
+```python
+df.plot(figsize=(10,8), 
+        title = "$H =A \ J_z + \delta \ J_x \  \cos (\omega t)$     (A=0.1, $\omega = 0.1$, $\delta=0.001$, $N=2$, $j=1$)   (Fig 6)")
+plt.xlabel("Time")
+plt.ylabel("Probability");
+```
+
+We can see in Fig 6 that we have successfully removed the flat line, corresponding to $j=0$, from our simulation with no changes to the evolution of the other states - as we would expect.
+
+Great! Now we are able to simulate an arbitrary number of TSS but just chaging a couple of numbers in our functions. Let's try it out with $N=6$.
+
+What $j$ values can we have for 6 TSS?
+
+```python
+j_vals(6)
+```
+
+Let's simulate the $j=3$ universe.
+
+```python
+J, jm_list = make_operators(6, 3)
+```
+
+```python
+result, ekets = simulate(A, delta, J, times, 0)
+```
+
+```python
+psi, P = make_p_psi_arrays(result.states, ekets)
+```
+
+```python
+df = pd.DataFrame(P, index=times)
+```
+
+```python
+df.plot(figsize=(10,8), 
+        title = "$H =A \ J_z + \delta \ J_x \  \cos (\omega t)$     (A=0.1, $\omega = 0.1$, $\delta=0.001$, $N=6$, $j=3$)   (Fig 7)")
+plt.xlabel("Time")
+plt.ylabel("Probability");
+```
+
+Fig 7 shows us something new. A system of 6 TSS can raise/lower its energy by an amount that is much greater than the individual TSS transition energy $\omega$. We see that the energy can changed by $6\omega$. It is able to do this by sequentially moving from one energy level to the next, each with a transition energy of $\omega$.
```

Comment:
so this is only due to the Delta*cos(omega t)? If that is removed, there is no transition, right?